packages/vite-extensions/vite-list-assets-required-for-offline-mode.ts

```diff
@@ -103,22 +103,23 @@ export const listAssetsRequiredForOfflineMode = ({
 	distDirectoriesToList,
 }: {
 	outputFile: string;
-	distDirectoriesToList: string[];
+	distDirectoriesToList: { path: string; destinationDirectory?: string }[];
 }) => {
 	return {
 		name: 'list-assets-required-for-offline-mode',
 		apply: 'build',
 		writeBundle({ dir: outputDir }: { dir: string }) {
 			const files = distDirectoriesToList.flatMap((dir) => {
-				const absoluteDirPath = join(outputDir, dir);
+				const destinationDirectory = dir.destinationDirectory || '';
+				const absoluteDirPath = join(outputDir, dir.path);
 				console.log(`Listing files in ${absoluteDirPath}`);
 				return listFiles(absoluteDirPath)
 					.map((file) => {
 						file = file.replace(absoluteDirPath, '');
 						if (file.startsWith('/')) {
-							return file;
+							return `${destinationDirectory}${file}`;
```

Comment:
any reason not to use `join()`?

Reply:
No, we should use `join`, updated it in this commit [10e3dfd](https://github.com/WordPress/wordpress-playground/pull/2841/commits/10e3dfda3cb6859c087342ea2165064234a04ffe).